woocommerce/woocommerce
# Apply spacing improvements to the Checkout block

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

This PR changes a lot of CSS to achieve the spacing requested in #46004 for both desktop and mobile.

Each checkout step now has a bottom margin of 48px. Previously, the titles of each step had padding/margin applied to the top however I moved it to the bottom of each checkout step.

Order notes, terms and conditions, and checkout actions all had to have some additional styling applied as they are outside of a "Checkout step".

As per the figma I moved the horizontal separator from between the Order Notes block and the Terms and Conditions block to be between the Terms and Conditions Block and the Checkout actions block (submit button). The separator is now tied to the checkout actions block and rendered above it.

#### Mobile 
| Before | After |
| ------ | ----- |
| <img width="351" alt="image" src="https://github.com/woocommerce/woocommerce/assets/5656702/f9bce722-cda9-40e9-b83c-bfe5d721bb17"> | <img width="354" alt="image" src="https://github.com/woocommerce/woocommerce/assets/5656702/3cbd46ed-c1b9-43a2-8fb0-e78f6a9f3e13"> |

#### Desktop

| Before | After |
| ------ | ----- |
| <img width="1617" alt="image" src="https://github.com/woocommerce/woocommerce/assets/5656702/6ae9c850-d6b2-4edd-82d9-8e6cfd318f72"> | <img width="1626" alt="image" src="https://github.com/woocommerce/woocommerce/assets/5656702/7b28f8ce-4348-4ab4-8273-6fce211457e0"> |





Closes #46004 .



### How to test the changes in this Pull Request:

> [!NOTE]
> Please repeat these steps in Chrome/Firefox/Safari with varying screen sizes. Please also test in different browsers on your mobile. Please also test with different themes. I tested with Storefront, Tsubaki, Twenty Twenty-Four, Tove.



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Reference figma: 2JyXNimI4oGeJZhgQNZE9V-fi-2072%3A31996

1. Add an item to your cart and go to the Checkout block.
2. Observe the vertical space between each checkout step. Ensure it is consistent, and it is 48px 
3. Check the Checkout block in the editor. Spacing should also be consistent here, although some padding has been added to various blocks to assist with highlighting them in the editor, so the checks here can be less strict, though all sections should look reasonably consistent and similar to what's rendered on the front-end.
4. In the editor, select the "Order notes" block and delete it. Ensure the spacing between elements is not affected by this and it still looks consistent.
5. Save the page and reload the Checkout block on the front end.
6. Again, check for vertical spacing consistency across all device sizes.
7. Return to the editor, remove the Checkout Terms and Conditions block.
7. Save the page and reload the Checkout block on the front end.
8. Again, check for vertical spacing consistency across all device sizes, paying special attention to the end of the form near the place order button.
9. Return to the editor, insert the order notes block above the Checkout button. Do not re-add the terms and conditions block yet.
10. Save the page and reload the Checkout block on the front end.
11. Check for vertical spacing consistency across all device sizes, paying special attention to the end of the form near the place order button. Be sure to check the "Add order note" box and ensure that the spacing is still fine with the textarea visible.
12. Go back to the editor, add the Checkout Terms block back above the place order button. Select it and turn on the `Require checkbox` option.
<img width="267" alt="image" src="https://github.com/woocommerce/woocommerce/assets/5656702/f4faf8b7-cf31-4fba-ae70-47b0a654802a">

13. Return to the Checkout block on the front end and reload. Try to check out without checking the Terms and Conditions box. Ensure the spacing is still fine when the checkbox is in its error state.
14. Test with Local pickup enabled/disabled and ensure everything is consistent.
15. Try various different configurations by modifying the block options of various inner blocks (use your creativity!) and ensure spacing is looking good in each configuration in both the editor and front-end.
16. Check the Cart block and compare it to trunk. Ensure no spacing changes have occurred.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
